test/cases/testdata/jsonbuiltins/test-json-marshal-with-options.yaml

```diff
@@ -0,0 +1,160 @@
+---
+cases:
+  - data: {}
+    modules:
+      - |
+        package test
+
+        p = x {
+          x := json.marshal_with_options([1234500000 + 67890, 1e6 * 2, 1e109 / 1e100], {"indent": "  ", "prefix": null})
+        }
+    note: jsonbuiltins/marshal_with_options-explicit
+    query: data.test.p = x
+    want_result:
+      - x: |-
+          [
+            1234567890,
+            2000000,
+            1000000000
+          ]
+  - data: {}
+    modules:
+      - |
+        package test
+
+        p = x {
+          x := json.marshal_with_options([1234500000 + 67890, 1e6 * 2, 1e109 / 1e100], null)
+        }
+    note: jsonbuiltins/marshal_with_options-nulled
```

Comment:
We should have a similar test for the empty `opts`object `{}`.

Reply:
7990d66433ae207a393648b720b234c24cf391e9 refactored this test accordingly.